tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -429,6 +429,7 @@ private void OnHostingHttpRequestInStart(object arg)
                     var scope = AspNetCoreRequestHandler.StartAspNetCorePipelineScope(tracer, httpContext, httpContext.Request, resourceName: string.Empty);
                     if (shouldSecure)
                     {
+                        CoreHttpContextStore.Instance.Set(httpContext);
```

Comment:
Does this ever need to get unset? I guess the answer is no because the AsyncLocal will only last for one request?

Reply:
Yes, my that's my assumption, from what I've read it seems to be true.